packages/aragon-wrapper/src/templates/index.js

```diff
@@ -26,49 +27,118 @@ const templates = {
-    ],
-  },
+      'neededSignatures' // number of signatures need, must be > 0 and <= signers.length
+    ]
+  }
 }
 
 const Templates = (web3, apm, from) => {
   const minGasPrice = web3.utils.toWei('20', 'gwei')
   const newToken = async (template, name) => {
-    const receipt = await template.methods.newToken(name, name).send({ from, gas: 4e6, gasPrice: minGasPrice })
-    return receipt.events.DeployToken.returnValues
+    const progress = new Subject()
+    const events = template.methods.newToken(name, name).send({ from, gas: 4e6, gasPrice: minGasPrice })
+
+    events.on('transactionHash', (transactionHash) => {
+      progress.next({
+        transaction: 'TOKEN',
+        status: 'SIGNED',
+        meta: {
+          transactionHash
+        }
+      })
+    }).on('receipt', ({ transactionHash, events }) => {
+      const tokenAddress = events.DeployToken.returnValues.token
+      progress.next({
+        transaction: 'TOKEN',
+        status: 'MINED',
+        meta: {
+          transactionHash,
+          address: tokenAddress
+        }
+      })
+      progress.complete()
+    }).on('error', (error, { transactionHash }) => {
+      progress.next({
+        transaction: 'TOKEN',
+        status: 'ERROR',
+        meta: {
+          transactionHash,
+          message: error.message
+        }
+      })
+      progress.complete()
+    })
+
+    return progress
   }
 
   const newInstance = async (template, name, params) => {
-    const receipt = await template.methods.newInstance(name, ...params).send({ from, gas: 6.6e6, gasPrice: minGasPrice })
-    return receipt.events.DeployInstance.returnValues
+    const progress = new Subject()
+    const events = template.methods.newInstance(name, ...params).send({ from, gas: 6.9e6, gasPrice: minGasPrice })
+
+    events.on('transactionHash', (transactionHash) => {
+      progress.next({
+        transaction: 'DAO',
+        status: 'SIGNED',
+        meta: {
+          transactionHash
+        }
+      })
+    }).on('receipt', ({ transactionHash, events }) => {
+      const daoAddress = events.DeployInstance.returnValues.dao
+      progress.next({
+        transaction: 'DAO',
+        status: 'MINED',
+        meta: {
+          transactionHash,
+          address: daoAddress
+        }
+      })
+      progress.complete()
+    }).on('error', (error, { transactionHash }) => {
+      progress.next({
+        transaction: 'DAO',
+        status: 'ERROR',
+        meta: {
+          transactionHash,
+          message: error.message
+        }
+      })
+      progress.complete()
+    })
+
+    return progress
   }
 
   return {
     newDAO: async (templateName, organizationName, params) => {
       const tmplObj = templates[templateName]
 
-      if (!tmplObj) throw new Error("No template found for that name")
-
-      const contractAddress = await apm.getLatestVersionContract(tmplObj.appId)
+      if (!tmplObj) {
+        throw new Error('No template found for that name')
```

Comment:
It feels weird that this would throw a normal error, but the next error would be captured in the observable.

It might be more intuitive for them both to have the same behaviour.

Reply:
Mhh, the other errors are transaction specified, so in my mind it would be weird that you could also get an error for the parameters you input